.circleci/config.yml

```diff
@@ -598,22 +719,152 @@ jobs:
+              exit 1
+            fi 
+            echo "release_version=[$release_version]"          
+            bumpversion_repo_user=${CIRCLE_PROJECT_USERNAME}
+            bumpversion_repo_name=${CIRCLE_PROJECT_REPONAME}
+            echo "bumpversion_repo_user:$bumpversion_repo_user"
+            echo "bumpversion_repo_name:$bumpversion_repo_name"
+
+            bump_version_message="[bump version $release_version]"
+            bump_version_pr_title="bump iOS sdk version to ${release_version}"
+            target_branch=test_develop
+            echo "export bumpversion_repo_user=$bumpversion_repo_user" >> $BASH_ENV
+            echo "export bumpversion_repo_name=$bumpversion_repo_name" >> $BASH_ENV
+            echo "export release_version=$release_version" >> $BASH_ENV
+            echo "export bump_version_message='${bump_version_message}'" >> $BASH_ENV
+            echo "export bump_version_pr_title='${bump_version_pr_title}'" >> $BASH_ENV
+            echo "export target_branch='${target_branch}'" >> $BASH_ENV
+            echo "bump_version_pr_title:$bump_version_pr_title"
+      - bump_version_pre
+      - run:
+          name: bump version
+          command: |
+            cp -R ${bumpversion_repo_name}/CircleciScripts/ CircleciScripts/
+            python3 CircleciScripts/bump_sdk_version.py "$(pwd)/${bumpversion_repo_name}" $release_version 
+      - bump_version_post
+  merge_to_master:
+    docker:
+      - image: circleci/android:api-27-alpha
+    environment:
+      JVM_OPTS: -Xmx1024m
+    steps:
+      - skip_job_unless_required
+      - checkout
+      - set_enviroment_variables
+      - run:
+          name: merge change to master
+          command: |
+            git config --local user.name AWS
+            git checkout test_master
+            git pull --rebase
+            commitlog=$(git log)
+            currentcommit="${CIRCLE_SHA1}"
+            echo "currentcommit=$currentcommit"
+            echo "$commitlog"
+            if [[ "$commitlog" =~ .*"commit $currentcommit".* ]] 
+            then
+               echo "The change is already merged"
+            else
+              echo "merge change from devlop to master"
+              currentcommit="${CIRCLE_SHA1}"
+              lastcommitmessage=$(git log --format=%B -n 1 ${currentcommit})
+
+              git merge  test_develop --no-edit --commit -m "[Merge develop to master by circleci] $lastcommitmessage"  
+              echo "push change"
+              git push -q https://${GITHUB_TOKEN}@github.com/${CIRCLE_PROJECT_USERNAME}/${CIRCLE_PROJECT_REPONAME}.git
+            fi
+  prepare_release_sdk:
+    docker:
+      - image: circleci/android:api-27-alpha
+    environment:
+      JVM_OPTS: -Xmx1024m
+    steps:
+      - skip_job_unless_required
+      - checkout
+      - set_enviroment_variables
+      - run:
+          name: trigger release_sdk if curremt commit is a bump version
+          command: |
+            git config --local user.name AWS
```

Comment:
since there are bunch of commands here, is it better to move them into a .sh file?

Reply:
done